### PR TITLE
add_isingxy_generator

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane_lightning/src/gates/cpu_kernels/GateImplementationsLM.hpp [hunk 1]
```diff
@@ -1111,6 +1112,38 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
...
+            const size_t i00 = ((k << 2U) & parity_high) |
+                               ((k << 1U) & parity_middle) | (k & parity_low);
+            const size_t i01 = i00 | rev_wire0_shift;
+            const size_t i10 = i00 | rev_wire1_shift;
+            const size_t i11 = i00 | rev_wire0_shift | rev_wire1_shift;
+            /*
+            const auto v01 = arr[i01];
+            arr[i01] = arr[i10];
+            arr[i10] = v01;
+            */
```
Comment (mlxd) [inline pennylane_lightning/src/gates/cpu_kernels/GateImplementationsLM.hpp:1138]:
```suggestion
```

File: pennylane_lightning/src/gates/cpu_kernels/GateImplementationsPI.hpp [hunk 2]
```diff
@@ -922,6 +923,30 @@ class GateImplementationsPI : public PauliGenerator<GateImplementationsPI> {
...
+        PL_ASSERT(wires.size() == 2);
+        const auto [indices, externalIndices] = GateIndices(wires, num_qubits);
+
+        for (const size_t &externalIndex : externalIndices) {
+            std::complex<PrecisionT> *shiftedState = arr + externalIndex;
+            /*
+            const auto v01 = shiftedState[indices[1]];
+            shiftedState[indices[1]] = shiftedState[indices[2]];
+            shiftedState[indices[2]] = v01;
+            */
```
Comment (mlxd) [inline pennylane_lightning/src/gates/cpu_kernels/GateImplementationsPI.hpp:940]:
```suggestion
```

**Review by mlxd — CHANGES_REQUESTED** [review]
Thanks @multiphaseCFD 
Just a small change requested.

File: pennylane_lightning/src/gates/cpu_kernels/GateImplementationsPI.hpp (hunk 2 above)
Comment (multiphaseCFD) [reply pennylane_lightning/src/gates/cpu_kernels/GateImplementationsPI.hpp:940]: Do you mean I need to remove it? Thanks

Comment (AmintorDusko) [reply pennylane_lightning/src/gates/cpu_kernels/GateImplementationsPI.hpp:940]: Yes. When with a red background, and without a text with a green background, it means a suggestion to remove the text. You can click Commit suggestion, or implement the change locally and commit and push the changes later.

**Comment by AmintorDusko** [comment]
Hi @multiphaseCFD, looks good so far!

File: pennylane_lightning/src/gates/cpu_kernels/GateImplementationsPI.hpp (hunk 2 above)
Comment (multiphaseCFD) [reply pennylane_lightning/src/gates/cpu_kernels/GateImplementationsPI.hpp:940]: Thanks @AmintorDusko  for the clarification!

File: pennylane_lightning/src/gates/cpu_kernels/GateImplementationsLM.hpp (hunk 1 above)
Comment (multiphaseCFD) [reply pennylane_lightning/src/gates/cpu_kernels/GateImplementationsLM.hpp]: Done!

**Comment by mlxd** [comment]
Maybe one more suggestion --- it would be great to add this gate to the Python tests also here:https://github.com/PennyLaneAI/pennylane-lightning/blob/a6f4c99f290ad53686ad3486676eeca22aaba3bd/tests/test_adjoint_jacobian.py#L627

**Comment by multiphaseCFD** [comment]
> 


> Maybe one more suggestion --- it would be great to add this gate to the Python tests also here:
> 
> https://github.com/PennyLaneAI/pennylane-lightning/blob/a6f4c99f290ad53686ad3486676eeca22aaba3bd/tests/test_adjoint_jacobian.py#L627

Added  and tested. Here is the screenshot of testing
image

**Comment by multiphaseCFD** [comment]
> Hi @multiphaseCFD, I think you still need to merge `origin/master` which is just updated, and make this PR ready for review. I will approve this when all tests pass.

Thanks @chaeyeunpark ! Will do it tomorrow!

**Review by mlxd — APPROVED** [review]

**Review by AmintorDusko — APPROVED** [review]
Nice work!

**Comment by chaeyeunpark** [comment]
Aha this is from rc branch.... It may complicate the branch tree, but I think we can merge it.

**Review by chaeyeunpark — APPROVED** [review]
Nice work!

**Pull request merged**